### DISCUSSION TITLE
How to specify your key for photo blocks

### DISCUSSION THREAD

**Discussion by GreenEyeInvestor** [igordanchenko/react-photo-album#110] [Q&A] [answered]
I have a problem with duplicate keys. As I understand it, the keys for the cells are formed from the src tag, this is all clear and correct, but I have content loading and sometimes there is a duplicate photo in the uploaded content, which leads to errors in the console

`Warning: Encountered two children with the same key, `image.jpeg`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
 `
```javascript
renderColumnContainer={({ columnContainerProps, children }) => (
                <div className={styles.photos__column}>{children}</div>
            )}
            renderContainer={({ containerProps, containerRef, children }: RenderContainerProps) => (
                <div ref={containerRef} className={styles.photos}>
                    {children}
                </div>
            )}
            renderPhoto={({ photo, wrapperStyle, renderDefaultPhoto }: any) => {
                return (
                    <ViewPhoto medium={photo} />
                )
            }}
``` 

Component ViewPhoto
```javascript
<div key={v4()}>
            <article className={`${styles.photoItem} ${styles.photoItemOverlay}`}  data-photo-modal-medium-id={medium._id}>
                <ImageComponent medium={medium} />
            </article>
        </div>
``` 
When specifying v4(), the library still uses its keys, how can I fix this?

**Comment by igordanchenko** [comment, marked as answer]
You should supply the `key` attribute as part of the photo object in this case.

https://react-photo-album.com/documentation#Photo

**Reply by GreenEyeInvestor** [reply]
Thanks a lot, somehow I missed this moment.
